benchmarks/benchmarks/bench_shape_base.py

```diff
@@ -93,7 +93,17 @@ def time_3d(self, n):
                 [self.a100, self.a101],
                 [self.a110, self.a111],
             ]
-        ])
+        ]
+        self.arr_list = [a.reshape(-1)
+                         for two_d in self.block
+                         for one_d in two_d
+                         for a in one_d]
+
+    def time_3d(self, n, mode):
+        if mode == 'block':
+            np.block(self.block)
+        else:
+            np.concatenate(self.arr_list)
```

Comment:
What's the goal here of testing concatenate within an `np.block` test?

Reply:
And why use a `mode` parameter rather than a separate benchmark function? Does this give a more useful asv result if the asv name is the same?

Some comments answering those questions in the code might be a good thing

Reply:
This is mostly so that we can directly compare the performance of `block` and `concatenate` on the same graph. I forget where, but you can see some graphs. Therefore, when creating the new algorithm, we should be able to stop optimizing once we are close to the performance of `concatenate`.

Reply:
It would look something like this
http://pandas.pydata.org/speed/dask/#array.TestSubs.time_subs